### ISSUE TITLE
Variable batch size error

### ISSUE THREAD

**Issue by terryrabinowitz** [snipsco/ntm-lasagne#22]
Hello. 
If I set the batch size to 'None' in the input layer of the model to allow different size batches, then I get the error: "ValueError: elements of reps must be scalars of integer dtype".   This error goes away when I directly connect this input layer to an output layer but appears when I try and connect the input layer to the ntm layer.

Thank you for the help and the implementation!
Terry

**Comment by tristandeleu** [comment]
Hello Terry,
Indeed, as the initializations of the memory and hidden states (for the recurrent controller) heavily rely on the size of the batches, I simply assumed they were given beforehand. Thank you for pointing that out!
This should be fixed in #23
Like the other recurrent layers from Lasagne (eg. `LSTMLayer`), you can access to the Theano variable corresponding to the size of the batches so that you can use `ReshapeLayer`
```python
l_input = InputLayer((None, None, size), input_var=input_var)
batch_size, seqlen, _ = l_input.input_var.shape
```

**Comment by terryrabinowitz** [comment]
Fantastic!  Thank you so much for the quick action!!
